api/lib/domain/usecases/create-user-and-reconcile-to-organization-learner-from-external-user.js

```diff
@@ -115,9 +117,37 @@ const createUserAndReconcileToOrganizationLearnerFromExternalUser = async functi
     }
   }
   const tokenUserId = userWithSamlId ? userWithSamlId.id : userId;
+
+  await _updateUserLastConnection({
+    requestedApplication,
+    userId: tokenUserId,
+    authenticationMethodRepository,
+    lastUserApplicationConnectionsRepository,
+    userLoginRepository,
+  });
+
   const accessToken = tokenService.createAccessTokenForSaml({ userId: tokenUserId, audience });
-  await userLoginRepository.updateLastLoggedAt({ userId: tokenUserId });
+
   return accessToken;
 };
 
 export { createUserAndReconcileToOrganizationLearnerFromExternalUser };
+
+async function _updateUserLastConnection({
+  requestedApplication,
+  userId,
```

Comment:
`userId` est plus important que `requestedApplication`, le mettre donc en premier :

```suggestion
async function _updateUserLastConnection({
  userId,
  requestedApplication,
```